src/go/rpk/pkg/cli/cluster/storage/list-mount.go

```diff
@@ -52,19 +54,33 @@ Use filter to list only migrations in a specific state
 				out.Exit(h)
 			}
 
-			pf, err := p.LoadVirtualProfile(fs)
+			p, err := p.LoadVirtualProfile(fs)
 			out.MaybeDie(err, "rpk unable to load config: %v", err)
-			config.CheckExitCloudAdmin(pf)
-			adm, err := adminapi.NewClient(cmd.Context(), fs, pf)
-			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+			config.CheckExitServerlessAdmin(p)
 
-			migrations, err := adm.ListMigrations(cmd.Context())
-			out.MaybeDie(err, "unable to list migrations: %v", err)
-			printDetailedListMount(p.Formatter, filterOptFromString(filter), rpadminMigrationStateToMigrationState(migrations), os.Stdout)
+			var migrations []rpadmin.MigrationState
+			if p.FromCloud {
+				cl, err := createDataplaneClient(p)
+				out.MaybeDieErr(err)
+
+				resp, err := cl.CloudStorage.ListMountTasks(cmd.Context(), connect.NewRequest(&dataplanev1alpha2.ListMountTasksRequest{}))
+				out.MaybeDie(err, "unable to list mount/unmount operations: %v", err)
+
+				if resp != nil {
```

Comment:
Is this conditional needed? `MaybeDie` exits on error. Can this return `nil, nil`?

Reply:
I am not sure, I was overly paranoid here to avoid a panic. `resp` is a `connect` wrapper of the response and I'm not sure if an empty list will produce an empty `resp`.

And I prefer being a little bit paranoid as we don't control the result at all. Happy to remove it if we are 100% sure this doesn't return `nil, nil`